### ISSUE TITLE
Default values for enableUnitTestCoverage and aggregateTestCoverage

### ISSUE THREAD

**Issue by ArcherEmiya05** [gmazzo/gradle-android-test-aggregation-plugin#144]
Is it possible to omit the call of `aggregateTestCoverage = false` for every release variant and any of its copy that uses `initWith`? IMHO it is very unlikely for someone to run unit test on release variant? However I also want to know if this is the way to opt in for a module in a multi-module project? My understanding is `aggregateTestCoverage` use to avoid duplicate unit test result in the report. I am planning to make a feature module available for public and putting `aggregateTestCoverage`  in one of its variants means that consumer of this library will be force to add this plugin too, good if it can be optional if possible?

**Comment by gmazzo** [comment]
This has been discussed already #58

**Comment by ArcherEmiya05** [comment]
> This has been discussed already #58

Sorry but how about `aggregateTestCoverage`? If I understand correctly from the sample app, its value is set in variants of a module?

**Comment by gmazzo** [comment]
Hi there, I'll try to clarify a few things:

The overall behavior of `aggregateTestCoverage` is documented [here](https://github.com/gmazzo/gradle-android-test-aggregation-plugin?tab=readme-ov-file#the-aggregatetestcoverage-dsl-extension)

But basically, when computing the value for a variant, it looks at each `aggregateTestCoverage` entry for its `BuildType` and any `Flavor`. Any explicit `true` will result in the module getting aggregated, otherwise any explicit `false` will prevent it, otherwise (all `null`) will result on `true` as well.

> Is it possible to prevent this without needing to explicitly specify aggregateTestCoverage = false in each module's release variant?

The spirit of this project was to link Gradle's Test Plugin with AGP, and therefore, if you have either Coverage (`enableUnitTestCoverage = true` enabled, default on `debug`) or Tests Results (there is no DSL on AGP to turn tests off as far I know for a variant), the plugin should aggregate them, if they are configured. This is intentional and opinionated.

The way to opt-out, the simplest way to avoid duplicating this, will be either using a convention plugin (in `buildSrc` folder for instance) or adding the following snippet in your root project:
```kotlin
allprojects {
    plugins.withId("com.android.base") {
        the<BaseExtension>().buildTypes {
            named("release") { aggregateTestCoverage = false }
            // or
            configureEach { aggregateTestCoverage = name == "debug" }
        }
    }
}
```
Make sure both AGP and this plugin are loaded in the root classloader (which is a good practice by the way) with a `plugins { id(xxx) apply false }`

> If the module is an open source, then the owner would need to add aggregateTestCoverage = false in its release variant and doing so will also force any consumer of that module to also know and include this plugin even if they don't want to.

Unless I've made some miss assumptions, this is not correct. "If the module is an open source" that will be an external dependency right? You'll be consuming it as a Maven dependency. 
This plugin only aggregates local Gradle modules, not their dependencies. 

Feel free to reach me further if you still have troubles with this ;)

**Comment by ArcherEmiya05** [comment]
> Hi there, I'll try to clarify a few things:
> 
> The overall behavior of `aggregateTestCoverage` is documented [here](https://github.com/gmazzo/gradle-android-test-aggregation-plugin?tab=readme-ov-file#the-aggregatetestcoverage-dsl-extension)
> 
> But basically, when computing the value for a variant, it looks at each `aggregateTestCoverage` entry for its `BuildType` and any `Flavor`. Any explicit `true` will result in the module getting aggregated, otherwise any explicit `false` will prevent it, otherwise (all `null`) will result on `true` as well.
> 
> > Is it possible to prevent this without needing to explicitly specify aggregateTestCoverage = false in each module's release variant?
> 
> The spirit of this project was to link Gradle's Test Plugin with AGP, and therefore, if you have either Coverage (`enableUnitTestCoverage = true` enabled, default on `debug`) or Tests Results (you can turn it off for instance by using `android.testOptions.unitTests.all { it.enabled = "Release" !in it.name }` DSL), the plugin should aggregate them, if they are configured.
> 
> This is intentional and opinionated. If it's available and configured to run, then it should be aggregated. You either don't have tests on `release` variants for instance, or you do have them and they should also run. The way to change this behavior, is to exclude them explicitly from aggregation.
> 
> The way to opt-out, the simplest way to avoid duplicating this, will be either using a convention plugin (in `buildSrc` folder for instance) or adding the following snippet in your root project:
> 
> ```kotlin
> allprojects {
>     plugins.withId("com.android.base") {
>         the<BaseExtension>().buildTypes {
>             named("release") { aggregateTestCoverage = false }
>             // or
>             configureEach { aggregateTestCoverage = name == "debug" }
>         }
>     }
> }
> ```
> 
> Make sure both AGP and this plugin are loaded in the root classloader (which is a good practice by the way) with a `plugins { id(xxx) apply false }`
> 
> > If the module is an open source, then the owner would need to add aggregateTestCoverage = false in its release variant and doing so will also force any consumer of that module to also know and include this plugin even if they don't want to.
> 
> Unless I've made some miss assumptions, this is not correct. "If the module is an open source" that will be an external dependency right? You'll be consuming it as a Maven dependency. This plugin only aggregates local Gradle modules, not their dependencies.
> 
> Feel free to reach me further if you still have troubles with this ;)

Tried the following in root level `build.gradle`:

```
import com.android.build.gradle.BaseExtension

// Top-level build file where you can add configuration options common to all sub-projects/modules.
buildscript {
    dependencies {
        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files

        // Add open source licenses notice
        classpath(libs.oss.licenses.plugin)
    }
}

plugins {
    alias(libs.plugins.android.application).apply(false)
    alias(libs.plugins.android.kotlin).apply(false)
    alias(libs.plugins.ksp).apply(false)
    alias(gms.plugins.google.services).apply(false)
    alias(firebase.plugins.crashlytics).apply(false)
    alias(libs.plugins.test.aggregation.results)
}

apply(plugin = "android-reporting")

// Since changing the report directory of instrumented tests result using `android-reporting` plugin will be much more difficult,
// we will instead change the report directory of unit tests result to align the two
tasks.testAggregatedReport {
    destinationDirectory.set(file("$rootDir/build/unitTest-results"))
}

allprojects {
    plugins.withId("com.android.base") {
        the<BaseExtension>().buildTypes {
            configureEach { aggregateTestCoverage = isDebuggable }
        }
    }
}
```

However I am getting the below error during Gradle sync.

`The value for this property is final and cannot be changed any further.`

It is pointing to this line `configureEach { aggregateTestCoverage = isDebuggable }`

**Comment by ArcherEmiya05** [comment]
> > However I am getting the below error during Gradle sync.
> > `The value for this property is final and cannot be changed any further.`
> 
> That I can fix it, bump to `2.2.1` ;)

Thanks a lot!!!
